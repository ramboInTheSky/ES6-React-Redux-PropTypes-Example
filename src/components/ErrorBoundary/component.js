import styled from 'styled-components';
import { Typography } from 'nhh-styles';

const Heading = styled(Typography.H3)`
  text-align: center;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
`;

export default Heading;
