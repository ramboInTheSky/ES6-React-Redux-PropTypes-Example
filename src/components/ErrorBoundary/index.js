import React from 'react';
import PropTypes from 'prop-types';

import Heading from './component';

class ErrorBoundary extends React.Component {
  state = { hasError: false };

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    if (this.state.hasError) {
      return this.props.messageOnScreen ? <Heading>{this.props.messageOnScreen}</Heading> : null;
    }
    return this.props.children;
  }
}
ErrorBoundary.defaultProps = {
  messageOnScreen: null,
};
ErrorBoundary.propTypes = {
  children: PropTypes.any.isRequired,
  messageOnScreen: PropTypes.string,
};

export default ErrorBoundary;
