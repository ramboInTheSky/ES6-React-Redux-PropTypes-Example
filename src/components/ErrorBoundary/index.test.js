import React from 'react';
import { shallow } from 'enzyme';

import ErrorBoundary from './';

describe('Error Boundary', () => {
  const props = {
    messageOnScreen: 'Something went wrong.',
  };

  const compositeComponent = (
    <ErrorBoundary messageOnScreen={props.messageOnScreen}>
      <p>this is a test!</p>
    </ErrorBoundary>
  );

  it('renders correctly', () => {
    expect(shallow(compositeComponent)).toMatchSnapshot();
  });

  it('generates a error message when an error is caught', () => {
    const component = shallow(compositeComponent);
    component.setState({
      hasError: true,
    });
    component.update();
    expect(component.children().text()).toEqual('Something went wrong.');
  });
});
