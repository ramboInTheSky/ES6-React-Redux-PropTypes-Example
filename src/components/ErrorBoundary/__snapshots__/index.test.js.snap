// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary renders correctly 1`] = `
<p>
  this is a test!
</p>
`;
